plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -632,11 +632,15 @@ private CpsScript parseScript() throws IOException {
             trusted = new CpsGroovyShellFactory(this).forTrusted().build();
             shell = new CpsGroovyShellFactory(this).withParent(trusted).build();
 
-            s = (CpsScript) shell.reparse("WorkflowScript",script);
+            s = (CpsScript) shell.reparse("WorkflowScript", script);
 
             for (Entry<String, String> e : loadedScripts.entrySet()) {
                 shell.reparse(e.getKey(), e.getValue());
             }
+        } catch (groovyjarjarasm.asm.MethodTooLargeException x) {
+            LOGGER.log(Level.SEVERE, "FAILED to parse WorkflowScript (the pipeline script) due to MethodTooLargeException: " + x.toString());
```

Comment:
Would be visible only to admins in the system log. To be useful, this would need to go into the build log. (Where it would be easily asserted by a test case.)

Reply:
Good point about a test case. Conjured up one that fits the current codebase (both master and this PR as it does not yet intrude into the build log).

Reply:
Git a bit lost in backtracking to how the "pretty message" can be shown in the build log, at least as really a message. It does not seem we have the `logger` context in a large part of the call stack where `parseScript()` is mentioned, so the best shot for visible logging seems to be to re-throw an exception with the desired message and this one as the linked cause.

Reply:
Updated below.